python/letsql/common/utils/dask_normalize.py

```diff
@@ -1,6 +1,25 @@
+from contextlib import contextmanager
+from unittest.mock import (
+    Mock,
+    patch,
+)
+
 import dask
+import toolz
+
 import letsql.common.utils.dask_normalize_expr  # noqa: F401
 import letsql.common.utils.dask_normalize_function  # noqa: F401
 import letsql.common.utils.dask_normalize_other  # noqa: F401
 
+
 dask.config.set({"tokenize.ensure-deterministic": True})
+
+
+@contextmanager
+def patch_normalize_token(*typs, f=toolz.functoolz.return_none):
+    with patch.dict(
+        dask.base.normalize_token._lookup,
+        {typ: Mock(side_effect=f) for typ in typs},
```

Comment:
Out of curiosity, why is the code using testing mocks?

Reply:
for `SnapshotStorage` to work, we need to remove some specific sources of non-determinism in `dask.base.normalize_token` but only during the context of specific invocations. `patch` fits this use and we already need this capability for tests.